updated_streamflow.py

```diff
@@ -0,0 +1,275 @@
+
+# Read and preprocess the USGS data
+df_plvo = utils.read_usgs_data(plvo_file, header_line=32)
+
+# Read and preprocess simulation data
+df_plvo_sim = utils.read_simulation_data(plvo_file_sim, header_line=6)
+
+# Merge df_plvo and df_plvo_sim
+df_plvo_merge = pd.merge(df_plvo, df_plvo_sim, on='datetime', how='inner')
+
+# Replace "CST" with "UTC" in the 'timezone' column
+
+df_plvo_merge['timezone'] = df_plvo_merge['timezone'].replace('CST', 'UTC')
+
+# Display the modified DataFrame
+print(df_plvo_merge.head())
+
+"""**Handling Missing Values**"""
+
+# Check for missing values in df_plvo_merged
+missing_values_plvo = df_plvo_merge.isnull().sum()
+
+# Print the number of missing values for each station
+
+print("\nNumber of missing values in df_plvo_merged:")
+print(missing_values_plvo)
+
+"""**Filter Date Range**"""
+
+# Convert 'datetime' column to datetime-like object
+df_plvo_merge['datetime'] = pd.to_datetime(df_plvo_merge['datetime'])
+
+# Filter the DataFrame for datetime values on "2007-01-01"
+filtered_df = df_plvo_merge[df_plvo_merge['datetime'].dt.date == pd.to_datetime("2007-01-01")]
+
+# Display the filtered DataFrame
+print(filtered_df)
+
+"""**Timeseries Plot**"""
+
+from plot_utils import plot_observed_vs_simulated_time_series
+
+# Plot observed vs simulated discharge time series for Pauls Valley station
+plot_observed_vs_simulated_time_series(df_plvo_merge, df_plvo_merge, 'Pauls Valley')
+
+"""**BenchMarking the Existing Physics Based Model**"""
+
+# Calculate NSE for Antlers station
+nse_Pauls_Valley = nse_utils.calculate_nse_for_station(df_plvo_merge, df_plvo_merge)
+
+
+print("NSE for Pauls Valley station:", nse_Pauls_Valley)
+
+"""**BenchMarking the Existing Physics Based Model**
+Based on the Validation time period
+
+
+---
+
+
+"""
+
+# Define the start and end dates
+start_date = '2016-01-01 00:00:00+00:00'
+end_date = '2018-12-31 18:00:00+00:00'
+
+# Create a new DataFrame within the specified timeframe
+df_plvo_merge_Val = df_plvo_merge[(df_plvo_merge['datetime'] >= start_date) & (df_plvo_merge['datetime'] <= end_date)]
+
+df_plvo_merge_Val
+
+# Calculate NSE for Antlers station
+nse_Pauls_Valley = nse_utils.calculate_nse_for_station(df_plvo_merge_Val, df_plvo_merge_Val)
+
+
+print("NSE for the validation period at Pauls Valley station:", nse_Pauls_Valley)
+
+df_plvo_merge
+
+df_plvo_merge.head()
+
+"""**Scaling - Standardization**"""
+
+# Standardize the discharge columns
+df_plvo_merge = standardization.standardize_discharge_columns(df_plvo_merge, ['Sim_Discharge', 'USGS_Discharge'])
+
+# Display the updated DataFrame
+df_plvo_merge.head()
+
+"""**Splitting the Data**"""
+
+df_plvo_merge['datetime'] = pd.to_datetime(df_plvo_merge['datetime'])
+
+train_start, train_end = '2007-01-01 06:00:00+00:00', '2015-12-31 18:00:00+00:00'
+val_start, val_end = '2016-01-01 00:00:00+00:00', '2018-12-31 18:00:00+00:00'
+test_start, test_end = '2019-01-01 00:00:00+00:00', '2020-12-31 12:00:00+00:00'
```

Comment:
Why are we starting and ending at different times? Does it not work if you just do this:
```suggestion
train_start, train_end = '2007-01-01', '2015-12-31'
val_start, val_end = '2016-01-01', '2018-12-31'
test_start, test_end = '2019-01-01', '2020-12-31'
```